Implement Fake Data Layer

shokeenashish4 opened the pull request:
This PR implements a fake data layer that allows a user to verify the App's behavior without having to utilize real data sources.https://github.com/shokeenashish4/figma_shopping_app/assets/150513800/58b10725-8b85-40f7-b3aa-654a22f7bacd

shokeenashish4 committed:
Introduce UI package

shokeenashish4 committed:
Introduce DI

shokeenashish4 committed:
Update button colors and onTap functionality on Gender Screen and add TextField validation for login and sign up screens

shokeenashish4 committed:
Implement Dashboard Screen with Side Drawer, Implement Reset Password feature and Make buttons on Gender Screen Selectable

shubhamsinghshubham777 commented on lib/data/products_repository.dart:
There's no code in this file yet. We can remove it.

shokeenashish4 replied:
done

shubhamsinghshubham777 commented on lib/ui/screens/dashboard_screen.dart:
```diff
@@ -0,0 +1,285 @@
+  int selectedIndex = 0;
+  final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey();
+
+  static const List<Widget> _widgetOptions = <Widget>[
```
Do we really need to make this variable `static`?

shokeenashish4 replied:
Removed the static and made it a 'final' list instead.

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/signup_onboarding_screen.dart:
```diff
@@ -0,0 +1,205 @@
+                showErrorMessage = false;
+              });
+
+              final wasSignUpSuccess = authRepo.signUp(
+                email: emailController.text,
+                password: passwordController.text,
+                username: usernameController.text,
+              );
+
+              if (await wasSignUpSuccess) {
+                Future.delayed(
+                  const Duration(seconds: 3),
+                  () {
+                    Navigator.of(context).push(
+                      MaterialPageRoute(
+                        builder: (context) => const GenderScreen(),
+                      ),
+                    );
+                  },
+                );
```
No need to use `Future.delayed` here. Use Future.delayed in `authRepo.signUp`'s implementation 👍🏻

shokeenashish4 replied:
done

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/social_onboarding_screen.dart:
```diff
@@ -42,23 +43,59 @@ class SocialButtonsSection extends StatelessWidget {
           text: "Facebook",
           iconPath: Assets.assetsFacebookLogo,
+          onTap: () {
+            Future.delayed(
+              const Duration(seconds: 3),
+              () {
+                Navigator.of(context).push(
+                  MaterialPageRoute(
+                    builder: (context) => const GenderScreen(),
+                  ),
+                );
+              },
+            );
+          },
```
While waiting for the response, we can show some kind of indication on the screen [maybe a circular progress indicator or `Loading...` text] (and this applies to all screens throughout the app) such that the user is never left hanging and he always gets some kind of response from the app.

shokeenashish4 replied:
done

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/social_onboarding_screen.dart:
```diff
@@ -42,23 +43,59 @@ class SocialButtonsSection extends StatelessWidget {
           text: "Google",
           iconPath: Assets.assetsGoogleLogo,
+          onTap: () {
+            Future.delayed(
+              const Duration(seconds: 3),
+              () {
+                Navigator.of(context).push(
+                  MaterialPageRoute(
+                    builder: (context) => const GenderScreen(),
+                  ),
+                );
+              },
+            );
```
We're using this exact same code snippet in both buttons. Better to extract it out as a function instead 👍🏻

shokeenashish4 replied:
done.

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/social_onboarding_screen.dart:
```diff
@@ -68,20 +105,24 @@ class SocialButtonsSection extends StatelessWidget {
       padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 5),
       child: ElevatedButton(
-        onPressed: () {},
+        onPressed: () {
+          onTap();
+        },
```
We can rewrite this to `onPressed: onTap`

shokeenashish4 replied:
done.

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/verification_code_screen.dart:
```diff
@@ -0,0 +1,123 @@
+          ),
+          textInputAction: inputAction,
+          keyboardType: TextInputType.number,
+          onChanged: (_) {
+            onSubmitted();
+          },
```
Optional: We can shorten this down to `onChanged: (_) => onSubmitted()`

shokeenashish4 replied:
done

shubhamsinghshubham777 commented on pubspec.yaml:
```diff
@@ -11,6 +11,8 @@ dependencies:
     sdk: flutter
   flutter_svg: ^2.0.9
   awesome_extensions: ^2.0.10
+  get_it: ^7.6.4
+  easy_sidemenu: ^0.5.0
```
Are we using any of these dependencies in the code till now? If not, we can remove it 👍🏻

shokeenashish4 replied:
removed

shubhamsinghshubham777 reviewed (changes requested):
Please check out the comments @shokeenashish4 and please also add demo videos/screenshots in this PR's description ✅

shokeenashish4 committed:
Show "Loading.." text after Social buttons are clicked.
Remove redundant code from signup_onboarding_screen.dart and verification_code_screen.dart
Remove products_repository.dart

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/login_screen.dart:
```diff
@@ -56,7 +56,12 @@ class _LoginScreenState extends State<LoginScreen> {
                 MaterialPageRoute(
-                  builder: (context) => const GenderScreen(),
+                  builder: (context) {
+                    const Center(
+                      child: CircularProgressIndicator(),
+                    );
```
Let's remove this as this LOC (Line of code) has no effect.

shubhamsinghshubham777 commented on lib/ui/screens/onboarding/new_pwd_screen.dart:
```diff
@@ -49,6 +49,7 @@ class _NewPwdScreenState extends State<NewPwdScreen> {
               const Duration(seconds: 2),
               () {
                 if (wasLoggedIn) {
+                  const Center(child: CircularProgressIndicator());
```
Same. Let's remove this as well.

shokeenashish4 committed:
Remove unnecessary line of code (CircularProgressIndicator)

shokeenashish4 commented:
Fixed the code with the changes you recommended
